cookbook/controller/service.rst

```diff
@@ -232,6 +234,87 @@ inject *only* the exact service(s) that you need directly into the controller.
    are valid, exactly how you want to organize your reusable code is up to
    you.
 
+Base Controller Methods and Their Service Replacements
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This list explains how to replace the convenience methods of the base
+controller:
+
+:method:`Symfony\\Bundle\\FrameworkBundle\\Controller\\Controller::createForm` (service: ``form.factory``)
+    .. code-block:: php
+
+        $formFactory->create($type, $data, $options);
+
+:method:`Symfony\\Bundle\\FrameworkBundle\\Controller\\Controller::createFormBuilder` (service: ``form.factory``)
+    .. code-block:: php
+
+        $formFactory->createBuilder('form', $data, $options);
+
+:method:`Symfony\\Bundle\\FrameworkBundle\\Controller\\Controller::createNotFoundException`
+    .. code-block:: php
+
+        new NotFoundHttpException($message, $previous);
+
+:method:`Symfony\\Bundle\\FrameworkBundle\\Controller\\Controller::forward` (service: ``http_kernel``)
+    .. code-block:: php
+
+        $httpKernel->forward($controller, $path, $query);
+
+:method:`Symfony\\Bundle\\FrameworkBundle\\Controller\\Controller::generateUrl` (service: ``router``)
+    .. code-block:: php
+
+       $router->generate($route, $params, $absolute);
+
+:method:`Symfony\\Bundle\\FrameworkBundle\\Controller\\Controller::getDoctrine` (service: ``doctrine``)
+
+    *Simply inject doctrine instead of fetching it from the container*
+
+:method:`Symfony\\Bundle\\FrameworkBundle\\Controller\\Controller::getUser` (service: ``security.context``)
+    .. code-block:: php
+
+        $user = null;
+        $token = $securityContext->getToken();
+        if (null !== $token && is_object($token->getUser())) {
+             $user = $token->getUser();
+        }
+
+:method:`Symfony\\Bundle\\FrameworkBundle\\Controller\\Controller::isGranted` (service: ``security.context``)
+    .. code-block:: php
+
+        $authorizationChecker->isGranted($attributes, $object);
+
+:method:`Symfony\\Bundle\\FrameworkBundle\\Controller\\Controller::redirect`
+    .. code-block:: php
+
+        return new RedirectResponse($url, $status);
```

Comment:
Same as [this comment](#r26005516) , why don't add FQCN for `RedirectResponse` class ?